src/jwtserviceauth-google.test.js

```diff
@@ -0,0 +1,145 @@
+// @ts-check
+'use strict'
+
+const expect = require('unexpected')
+const { createTestHttpServer } = require('./testutils')
+const { rsaPrivateKey, rsaPublicKey } = require('./testresources')
+
+const querystring = require('querystring')
+const { JwtServiceAuth, JwtUtils } = require('./index')
+
+const r2 = require('r2')
+const curl = require('url')
+
+let googleKeyFileData = {
+  type: 'service_account',
+  project_id: 'test-project',
+  private_key_id: '76d81ae69ce620a517b140fc73dbae61e88b34bc',
+  private_key: rsaPrivateKey,
+  client_email: 'buildstatus@nversion-168820.iam.gserviceaccount.com',
+  client_id: '123456789123456789123',
+  auth_uri: 'https://accounts.google.com/o/oauth2/auth',
+  token_uri: 'https://accounts.google.com/o/oauth2/token',
+  auth_provider_x509_cert_url: 'https://www.googleapis.com/oauth2/v1/certs',
+  client_x509_cert_url:
+    'https://www.googleapis.com/robot/v1/metadata/x509/servicename%40test-project.iam.gserviceaccount.com'
+}
+
+const pubKeys = {
+  'buildstatus@nversion-168820.iam.gserviceaccount.com': {
+    '76d81ae69ce620a517b140fc73dbae61e88b34bc@RS256': rsaPublicKey
+  }
+}
+
+describe('JwtServiceAuth', () => {
+  let [httpServer, listenPromise] = createTestHttpServer((req, res) => {
```

Comment:
Shouldn't the createTestHttpServer definition be in the before loop also? And then reference the previous current scope?